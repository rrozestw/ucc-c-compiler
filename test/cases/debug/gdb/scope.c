// RUN: %debug_scope %s | %stdoutcheck %s

char a;

f(a,b,c,d,e,f,
	g,h,i,j,k,l,m,n,
	pt, fp)

	float g, h, i, j, k, l, m, n;

	char *pt;
	float fp;

{
	char *pt_local = pt + 1;

	printf("%p %f\n", pt, fp);

	float *pfp = &fp;
	if(*pfp != fp)
		abort();

	{
		extern char a;
		if(pt_local != &a + 1)
			abort();
	}
}

main()
{
	int local1 = 1;

	printf("&a = %p\n", &a);

	int local5 = 5;

	f(1,2,3,4,local5,6,
		1,2,3,4,5,6,7,8,
		&a, 3);
}

// global a:
// STDOUT: a 5 41
// args, locals, etc:
// STDOUT: a<1> 5 30
// STDOUT: a<2> 25 28
// STDOUT: b 5 30
// STDOUT: c 5 30
// STDOUT: d 5 30
// STDOUT: e 5 30
// STDOUT: f 5 30
// STDOUT: fp 5 30
// STDOUT: g 5 30
// STDOUT: h 5 30
// STDOUT: i 5 30
// STDOUT: j 5 30
// STDOUT: k 5 30
// STDOUT: l 5 30
// STDOUT: local1 30 41
// STDOUT: local5 32 41
// STDOUT: m 5 30
// STDOUT: n 5 30
// STDOUT: pfp 15 28
// STDOUT: pt 5 30
// STDOUT: pt_local 5 30
