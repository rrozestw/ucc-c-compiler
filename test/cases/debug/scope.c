// RUN: %debug_check %s

struct A // not output - unused
{
	enum E
	{
		X123, Y123, Z123
	} e;
	int i, j;
};

struct B
{
	enum E2
	{
		X321, Y321, Z321
	} e;
	int i, j;
} b;

static int f(int i)
{
	static int si = 3;
	return i + si;
}

main(int argc, char *argv[])
{
	struct A
	{
		int i, j;
	} x = {
		f(52),
		3
	};
}
