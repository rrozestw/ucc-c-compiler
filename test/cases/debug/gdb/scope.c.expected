# check variables are as we expect and we are on the right line
33		int local1 = 1;
/\\$1 = 0x[0-9a-f]+ .*""/
35		printf("&a = %p\n", &a);
37		int local5 = 5;
local5 = 5
local1 = 1
16		char *pt_local = pt + 1;
20		float *pfp = &fp;
$2 = 3
$3 = 1
# this ensures that { extern a; ... } replaces local a:s with global a:s
/\\$4 = 0 '\0+'/
$5 = 1
/&a = 0x[0-9a-f]+/
